### DISCUSSION TITLE
Proper documentation???

### DISCUSSION THREAD

**Discussion by tamis-laan** [nats-io/nats.go#1626] [Q&A] [answered]
The readme documentation does not match the latest version of the client. For example the method `js.Stream(ctx, "foo")` does not exist and there is no support for `context` at all. I'm running version:
```
github.com/nats-io/nats.go v1.34.1
```
Where can I find the latest documentation?

EDIT:

Looks like I'm using the "old" interface as I have methods available such as `js.AddStream`. Which is very confusing given the fact I'm using the latest version of the library!? I thought maybe I'm using the wrong import? But `"github.com/nats-io/nats.go"` should be correct as it matches the examples in the README.

**Comment by piotrpio** [comment, marked as answer]
Hello @tamis-laan. As you discovered already, you were using the old API from `nats` package, whereas the new API is in `jetstream` package. We are aware that this may be confusing and eventually we may deprecate the old API, however since many users depend on it we are supporting both.

I think it's a good idea to put an easily visible disclaimer about the difference and where to find the new API.

**Reply by tamis-laan** [reply]
Yeah, I figured it out shortly after. Wouldn't it be more logical to use something like the following convention:
```
go get github.com/nats-io/nats.go/v1
go get github.com/nats-io/nats.go/v2
```
I see this being used on more go packages.

**Reply by Jarema** [reply]
Sure, this is often used.
However this is just a new version of JetStream client API, not the whole client.
Besides, we anyway wanted to separate jetstream into separate package.

**Reply by tamis-laan** [reply]
>  Besides, we anyway wanted to separate jetstream into separate package.

I think that would make things much clearer and cleaner. Then both packages can be versioned independently.
